src/Illuminate/Auth/Passwords/PasswordBroker.php

```diff
@@ -95,25 +121,27 @@ public function sendResetLink(#[\SensitiveParameter] array $credentials, ?Closur
      */
     public function reset(#[\SensitiveParameter] array $credentials, Closure $callback)
     {
-        $user = $this->validateReset($credentials);
+        return $this->timebox->call(function ($timebox) use ($credentials, $callback) {
+            $user = $this->validateReset($credentials);
 
-        // If the responses from the validate method is not a user instance, we will
-        // assume that it is a redirect and simply return it from this method and
-        // the user is properly redirected having an error message on the post.
-        if (! $user instanceof CanResetPasswordContract) {
-            return $user;
-        }
+            // If the responses from the validate method is not a user instance, we will
+            // assume that it is a redirect and simply return it from this method and
+            // the user is properly redirected having an error message on the post.
+            if (! $user instanceof CanResetPasswordContract) {
+                return $user;
+            }
 
-        $password = $credentials['password'];
+            $password = $credentials['password'];
 
-        // Once the reset has been validated, we'll call the given callback with the
-        // new password. This gives the user an opportunity to store the password
-        // in their persistent storage. Then we'll delete the token and return.
-        $callback($user, $password);
+            // Once the reset has been validated, we'll call the given callback with the
+            // new password. This gives the user an opportunity to store the password
+            // in their persistent storage. Then we'll delete the token and return.
+            $callback($user, $password);
 
-        $this->tokens->delete($user);
+            $this->tokens->delete($user);
 
-        return static::PASSWORD_RESET;
+            return static::PASSWORD_RESET;
```

Comment:
When the reset is successful you can do an early return I think.
```suggestion
            $timebox->returnEarly();

            return static::PASSWORD_RESET;
```

Reply:
Good point, yes, I think it's safe to do that here. 👍

I couldn't use it in `sendResetLink()` as there were only two states, but `reset` does have three.